### DISCUSSION TITLE
Updated baremetal compile options for Armv8.2a+sve and RVV

### DISCUSSION THREAD

**Discussion by mrgabich** [OpenMathLib/OpenBLAS#4342] [Q&A]
Hi dear, 

Is there some updated tutorial about compiling OpenBLAS targeting baremetal workloads?
If not, how hard it would be to generate 2 versions: 1 for Armv8.2-a+sve and 1 for RV64GCV?

Many Thanks
Geancarlo Abich

**Comment by StuartIanNaylor** [comment]
Due to my poor compile skills that never seems to go well in Ubuntu I just used sudo apt-build install openblas...
I noticed even though native was selected on a Cortex a55/a76 rk3588 a of the architecture and mtune seems to be hardcoded where a53/a72 was processed.
Much did tune to a55/a76 but it looks like some is hardcoded to just base aarch architecture.
Dunno if apt-build might be easier for you, but wondering how much is optimised to make use of the mat/mul and extra instructions of Armv8.2+ ?

**Comment by martin-frbg** [comment]
If you are going for baremetal, I expect the critical step will be providing your own functions for memory management (and setting EMBEDDED=1 to disable the related code and inclusion of system headers in the OpenBLAS source). There is a (very) short wiki page for ARM Cortex M that touches on that. I must admit that I have never done this myself, and I am not aware if anybody has ever tried it with RISCV (though I think the procedure will be essentially the same)
